src/core/thread/mle_router.cpp

```diff
@@ -1706,13 +1706,6 @@ otError MleRouter::HandleParentRequest(const Message &aMessage, const Ip6::Messa
 
     child = FindChild(macAddr);
 
-    if (child != NULL && !child->IsFullThreadDevice())
-    {
-        // Parent Request from a MTD child means that the child had detached. It can be safely removed.
-        RemoveNeighbor(*child);
-        child = NULL;
-    }
```

Comment:
Do we expect a re-attach after a "Parent Request"? Or the child is assumed to stay attached and can send a "Parent Request" to juct scheck for possible new/better parents?

What if the child is re-attaching and then chooses a different parent. If I am not mistaken, we (as current parent) would then keep the child in "VALID" state and keep the queued messages for it till the timeout.  Would we  also continue to respond to address query for it (same as its new parent)?